src/ci_providers/provider_bitbucket.js

```diff
@@ -0,0 +1,79 @@
+const childProcess = require('child_process')
+
+function detect(envs) {
+  return envs.CI && envs.BITBUCKET_BUILD_NUMBER
+}
+
+function _getBuild(inputs) {
+  const { args, envs } = inputs
+  return args.build || envs.BITBUCKET_BUILD_NUMBER
+}
+
+// eslint-disable-next-line no-unused-vars
+function _getBuildURL(inputs) {
+  return ''
+}
+
+function _getBranch(inputs) {
+  const { args, envs } = inputs
+  return args.branch || envs.BITBUCKET_BRANCH || ''
+}
+
+function _getJob(envs) {
+  return envs.BITBUCKET_BUILD_NUMBER
+}
+
+function _getPR(inputs) {
+  const { args, envs } = inputs
+  return args.pr || envs.BITBUCKET_PR_ID || ''
+}
+
+function _getService() {
+  return 'bitbucket'
+}
+
+function getServiceName() {
+  return 'Bitbucket'
+}
+
+function _getSHA(inputs) {
+  const { args, envs } = inputs
+  let commit = envs.BITBUCKET_COMMIT
+
+  if (commit && commit.length === 12) {
+    commit = childProcess.execSync(
```

Comment:
Please create a validate method in src/helpers/validate.js and sanitize prior to passing any envs or args  to any ChildProcess methods.

@drazisil-codecov to refactor when feeling better.

Reply:
This is being re-implemented in #186